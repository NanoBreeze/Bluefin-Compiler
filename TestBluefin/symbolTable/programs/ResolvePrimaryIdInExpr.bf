int a;
float b;
void f() {
	a = 5;
	a = b;
}
bool g(float a) {
	b = 35.2;
	a = 34.6;
}

int main() {
	f();
	g(4);
	c = 9;
}