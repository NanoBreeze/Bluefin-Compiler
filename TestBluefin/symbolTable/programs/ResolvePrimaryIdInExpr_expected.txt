resolve - int - c_builtinType - t_int
declare - a - c_var - t_int
resolve - float - c_builtinType - t_float
declare - b - c_var - t_float
resolve - void - c_builtinType - t_void
declare - f - c_function - t_void
enterScope - Level 1
exitScope - Level 1
resolve - bool - c_builtinType - t_bool
declare - g - c_function - t_bool
enterScope - Level 1
resolve - float - c_builtinType - t_float
declare - a - c_var - t_float
exitScope - Level 1
resolve - int - c_builtinType - t_int
declare - main - c_function - t_int
enterScope - Level 1
exitScope - Level 1
resolve - a - c_var - t_int
resolve - a - c_var - t_int
resolve - b - c_var - t_float
resolve - b - c_var - t_float
resolve - a - c_var - t_float
resolve - f - c_function - t_void
resolve - g - c_function - t_bool
resolve - c - UNRESOLVED
